config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java

```diff
@@ -68,6 +71,33 @@ private static void validateTotalDelay(List<Step> steps) {
             throw new IllegalArgumentException("The total delay specified is " + Duration.ofSeconds(totalDelaySeconds) +
                                                " but max 24 hours is allowed");
     }
+
+    /** Throw an IllegalArgumentException if any production zone is declared multiple times */
+    private static void validateZones(List<Step> steps) {
+        // Collect both non-parallel and parallel zones
```

Comment:
I think this could be made simpler, but ok.

Reply:
Agreed, it's a bit inelegant. Any concrete suggestions?